### DISCUSSION TITLE
What is the correct way to setFocus on a RadioGroup

### DISCUSSION THREAD

**Discussion by kachmul2004** [react-hook-form/react-hook-form#8153] [Q&A]
What's the right way to setFocus on a required RadioGroup which doesn't have a default option on render? Where should the ref go? I tried to put the ref on several components inside the render and the only one that seemed to work was when in add inputRef={ref} on the Radio options themselves, but it worked randomly/inconsistently on PC, but on mobile phone it doesn't even seem to work. 

My component looks like this:

`
import {
  Divider,
  FormControl,
  FormControlLabel,
  FormHelperText,
  Radio,
  RadioGroup,
  Typography,
} from "@mui/material";
import { subTextColor } from "./constants";
import { Controller, useFormContext } from "react-hook-form";

export const SatisfactionRadioButtons = ({ proponent, name }) => {
  const { control } = useFormContext();
  /*-----Questions Strings-----*/
  const satisfactionLevelDescriptions = [
    { value: "At Risk of Losing them" },
    { value: "Disappointed, but we are working on improvements" },
    { value: "Satisfied, no complaints but no praise either" },
    { value: "Happy, often praising DreamSave/DSL" },
    {
      value: `${proponent} recommending DreamSave/DSL to friends and colleagues`,
    },
  ];

  return (
    <>
      <Typography variant="h5">
        {proponent === "Partner"
          ? `How `
          : `are you`}
      </Typography>
      <Controller
        render={({
          field: { onChange, value, ref },
          fieldState: { error, invalid },
        }) => (
          <FormControl>
            <RadioGroup value={value} onChange={onChange} ref={ref}>
              <FormHelperText error={invalid}>
                {invalid && error.message}
              </FormHelperText>
              {satisfactionLevelDescriptions.map((level, index) => (
                <FormControlLabel
                  style={subTextColor}
                  key={index}
                  value={level.value}
                  control={
                    <Radio
                      key={index}
                      sx={{
                        "& .MuiSvgIcon-root": {
                          fontSize: 16,
                        },
                      }}
                    />
                  }
                  label={level.value}
                />
              ))}
            </RadioGroup>
          </FormControl>
        )}
        name={name}
        control={control}
      />
      <Divider sx={{ mb: 2, mt: 2 }} />
      <Typography variant="h5">Please explain with more details:</Typography>
      <Typography variant="string" sx={subTextColor}>
        {proponent === "Partner"
          ? `Hello`
          : `World?`}
      </Typography>
    </>
  );
};

`

**Comment by bluebill1049** [comment]
provide a codesandbox.

**Comment by Moshyfawn** [comment]
I'd suggest providing a default value for your radio group as it's not recommended to have an empty radio group.

From [the MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)

> It's fairly uncommon to actually want to allow the form to be submitted without any of the radio buttons in a group selected, so it is usually wise to have one default to the checked state. See [Selecting a radio button by default](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#selecting_a_radio_button_by_default) below.

Also, from [the MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio):
> Radio buttons don't participate in constraint validation; they have no real value to be constrained.

As to where to pass your `ref` callback, your best bet is to pass it to one of your radio buttons.
